src/developer-quickstart.md

```diff
@@ -0,0 +1,441 @@
+# Developer Quickstart
+
+This guide will walk developers through writing a smart contract in Sway, a simple test, deploying to Fuel, and building a frontend.
+
+Before we begin, it may be helpful to understand terminology that will used throughout the docs and how they relate to each other:
+
+- **Fuel**: the Fuel blockchain.
+- **FuelVM**: the virtual machine powering Fuel.
+- **Sway**: the domain-specific language crafted for the FuelVM; it is inspired by Rust.
+- **Forc**: the build system and package manager for Sway, similar to Cargo for Rust.
+
+## Understand Sway Program Types
+
+There are four types of Sway programs:
+
+- `contract`
+- `predicate`
+- `script`
+- `library`
+
+Contracts, predicates, and scripts can produce artifacts usable on the blockchain, while a library is simply a project designed for code reuse and is not directly deployable.
+
+|             | deployable on the blockchain:       | can have state:      | callable on the blockchain: | designed for code reuse: |
+|-------------|-------------------------------------|----------------------|-----------------------------|--------------------------|
+| contract  | ✅                                  | ✅                   | ✅                          | ❌                        |
+| predicate | ✅                                  | ❌                    | ❌                           | ❌                        |
+| script    | ❌                                   | ❌                    | ❌                           | ❌                        |
+| library   | ✅ (via a contract or predicate) | ✅ (via a contract) | ❌                           | ✅                       |
```

Comment:
Actually, libraries can't do anything with state currently 😂. https://github.com/FuelLabs/sway/issues/2585

Reply:
Ok, I thought so, had asked in Slack but wasn't sure. Removing that part